### PR TITLE
Correctly serialize IList and IDictionary in CreateToken

### LINKED ISSUE #2245: Dictionary claims added to SecurityTokenDescriptor.Claims are no longer correctly serialized in 7.0.0-preview3
Dictionaries (but it's likely other types of complex claims are impacted) are no longer correctly serialized in 7.0.0-preview3:

```csharp
var dictionary = new Dictionary<string, string[]>
{
    ["prop1"] = new[] { "value1", "value2" }
};

var descriptor = new SecurityTokenDescriptor
{
    Claims = new Dictionary<string, object>
    {
        ["a"] = dictionary
    },

    SigningCredentials = new SigningCredentials(new RsaSecurityKey(RSA.Create(2048)), SecurityAlgorithms.RsaSha256)
};

var handler = new JsonWebTokenHandler();

var token = handler.CreateToken(descriptor);
```

7.0.0-preview:

```json
{
  "a": {
    "prop1": [
      "value1",
      "value2"
    ]
  },
  "exp": 1692706803,
  "iat": 1692703203,
  "nbf": 1692703203
}
```

7.0.0-preview3

```json
{
  "a": "System.Collections.Generic.Dictionary`2[System.String,System.String[]]",
  "exp": 1692706612,
  "iat": 1692703012,
  "nbf": 1692703012
}
```

Complex claims should be correctly serialized or an exception should be thrown, but using the CLR type name as the claim value in the generated token seems like a terrible choice.

/cc @brentschmaltz @jennyf19

### PR DESCRIPTION
**Pull request by keegan-caruso**
Fixes #2245

### CODE REVIEW

**Review by jmprieur — APPROVED** [review]
LGTM

File: test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenHandlerTests.cs [hunk 1]
```diff
@@ -63,6 +63,160 @@ public void JsonWebTokenHandler_CreateToken_SameTypeMultipleValues()
...
+                },
+                new ComplexTypeSerializationTheoryData
+                {
+                    ClaimEntry =new KeyValuePair<string, object>(
+                        "keyH",
+                        new Dictionary<string, object>
+                        {
+                            ["prop10"] = new Dictionary<string, bool>
+                            {
+                                ["prop12"] = true,
```
Comment (jennyf19) [inline test/Microsoft.IdentityModel.JsonWebTokens.Tests/JsonWebTokenHandlerTests.cs]: nit: should be "prop11", "prop12"  if I'm following your pattern correctly.

**Review by jennyf19 — APPROVED** [review]
:shipit:

**Comment by jennyf19** [comment]
This is great @keegan-caruso ...nice to see the additional test coverage.

**Comment by keegan-caruso** [comment]
closing in favor of #2268

**Pull request closed without merging**